src/App.tsx

```diff
@@ -1,17 +1,122 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import TodoList from './components/TodoList/TodoList';
 
 import users from './api/users';
+import todos from './api/todos';
+
+interface Todo {
+  userId: number;
+  id: number;
+  title: string;
+  completed: boolean;
+  userLink?: {
+    name: string;
+    email: string;
+  }
+}
+
+let todoData: Todo[] = todos.map(todo => {
+  return {
+    ...todo,
+    userLink: users.find(user => user.id === todo.userId),
+  };
+});
+
+const createTodo = (text: string, id: number) => {
+  const newTodo = {
+    userId: id,
+    id: todoData.length + 1,
+    title: text,
+    completed: false,
+    userLink: users.find(user => user.id === id),
+  };
+
+  todoData = [
+    ...todoData,
+    newTodo,
+  ];
+};
```

Comment:
move this function inside the App component, also add todos to state and each time you create a new todo update the state, not array todoData